.circleci/config.yml

```diff
@@ -13,12 +13,15 @@ jobs:
       - run: git submodule update --remote --force
       - run: cd wp-e2e-tests && git checkout origin/${E2E_BRANCH-master}
       - restore_cache:
-          key: v1-{{ checksum "wp-e2e-tests/package-lock.json" }}
-      - run: cd wp-e2e-tests && npm install
+          keys: 
+            - v1-npmcache-{{ checksum "wp-e2e-tests/.nvmrc" }}-{{ checksum "wp-e2e-tests/package-lock.json" }}
+            - v1-npmcache-{{ checksum "wp-e2e-tests/.nvmrc" }}
+            - v1-npmcache
```

Comment:
is this line meant to be here? it's not in the `wp-e2e-tests-canary` config

Reply:
This allows the cache to be persisted when Node versions change.

@blowery did you intentionally include this? Makes me a bit nervous and I'd expect to miss the other two keys and match this very infrequently.

Reply:
@sirreal it was intentional, but we can drop it. it would only get hit the first time we switch node versions.

Reply:
This would maintain the cache indefinitely. I'm actually a bit curious to see how that plays out.

I wonder what the growth of that cache will be like and whether the restore/save steps will be noticeably slower just due to size.

No need to remove it on my part.

Reply:
the cache won't grow, circle's caches are all immutable. once created for a key, they never change.

Reply:
An individual _CircleCI_ cache won't grow, but each new _CircleCI_ cache will be larger. The _npm_ cache will always be persisted, so any new _CircleCI_ cache will include the `npm` cache plus caches for any new modules.

Reply:
ah, i see where you're going. for the switch case, we restore the original npm cache (from the v1-npmcache key) then install all the new stuff and save the cache under a "full" key.

Any one circle CI cache will only have the `v1-npmcache` (or at worst, that plus the `v1-npmcache-{{ checksum "wp-e2e-tests/.nvmrc"}}`) bits in addition to the full set. 

Over time the `v1-npmcache` is probably just dead weight. 

But I don't see any weird unbounded growth occurring...